Restguru

tommysullivan opened the pull request:
This PR adds a web server with a starting example of how restguru might work, and readme instructions on how to run that locally.
<img width="1680" alt="Screen Shot 2019-09-04 at 12 24 20 PM" src="https://user-images.githubusercontent.com/1911377/64299350-ca898980-cf2d-11e9-9531-249d0d7d8aaf.png">
<img width="1680" alt="Screen Shot 2019-09-04 at 12 21 51 PM" src="https://user-images.githubusercontent.com/1911377/64299362-d4ab8800-cf2d-11e9-8495-a04ada684ffc.png">
<img width="1680" alt="Screen Shot 2019-09-04 at 11 07 54 AM" src="https://user-images.githubusercontent.com/1911377/64299376-e2610d80-cf2d-11e9-96f6-e591c3945bbb.png">
<img width="1680" alt="Screen Shot 2019-09-03 at 8 14 36 PM" src="https://user-images.githubusercontent.com/1911377/64299408-f7d63780-cf2d-11e9-8b06-889f6764e731.png">
![Screen Shot 2019-09-07 at 2 19 29 PM (2)](https://user-images.githubusercontent.com/1911377/64480339-91922480-d17a-11e9-9ef6-f62cd1739120.png)

tommysullivan commented:
@bob-bins this isnt ready, i see the branch has some extra stuff in it, will ping u when its ready.

tommysullivan commented:
@bob-bins this is ready!

bob-bins commented on docs/clientDev.md:
```diff
@@ -1,6 +1,15 @@
+# Prerequisites
+
+Ensure the [api dev server](./serverDev) is running, since the webpack-dev-server will proxy all non-frontend requests
+thereto; if it is not running, 
```
you forgot to complete the sentence

bob-bins commented on src/frontEnd/RestGuru/RestGuru.tsx:
```diff
@@ -0,0 +1,172 @@
+interface A {
+    a: number
+}
+export default A
```
lines 4-8 are leftover from testing?

bob-bins commented on src/frontEnd/nav/SideNav.tsx:
```diff
@@ -66,6 +66,10 @@ export const SideNavUnstyled = ({navTo, isSideNavOpen, isSideNavExpanded, onSide
               <ListItemText primary="Castle Risk" />
             </ListItem>
+            <ListItem button onClick={() => navTo("/rest-guru") }>
+              <ListItemIcon><DonutSmallIcon /></ListItemIcon>
```
change the icon! 😛

tommysullivan replied:
u change it! lol

bob-bins commented on src/utils/debugging/log.ts:
```diff
@@ -0,0 +1,6 @@
+
+export const log:Log = (...args) => {
+    console.log(...args)
+    return args[args.length - 1]
```
is this function supposed to have a `return` value?

bob-bins replied:
nevermind, i can see further below that this is written as intended. Though it's a bit odd to me why a log function would return something, and that that something is the last argument passed in... but forgetting about that for a moment, it would be good to handle the case where no args are passed

tommysullivan replied:
chatted on this, shoudl be good!

bob-bins commented on src/frontEnd/RestGuru/contentType.md:
```diff
@@ -0,0 +1,82 @@
+        "text/html",
+        "application/rest.guru;href=http://rest.guru/contentTypes/text/plain",
+        "application/rest.guru;href=http://rest.guru/contentTypes/text/html",
+        "application/rest.guru;href=http://rest.guru/contentTypes/typescript,
```
```suggestion
        "application/rest.guru;href=http://rest.guru/contentTypes/typescript",
```

bob-bins commented:
If you are adamant about having line breaks in the `.md` files, it would help to restrict the lines to 118 characters long, so it will render nicely in Github.

tommysullivan commented:
> If you are adamant about having line breaks in the `.md` files, it would help to restrict the lines to 118 characters long, so it will render nicely in Github.

updated to be 118 characters long! it is typical for md files to have explicit line breaks so knowing that I am hesitant to not put them; though I think with those of us who actually look at them, we look at them in tools where an atypical lack of line breaks would look very nice, better on average than with explicit line breaks perhaps!

in any case, i updated the vs code settings to also display line breaks at 118 chars so that we can more readily see if we are putting them in awkward places that would look bad on github. :)

tommysullivan commented:
@bob-bins what do u think, mergeable!?!?! :)

tommysullivan commented:
@bob-bins can you sign off on this so i can merge without an override?
